CLD with ARC

atolosadelgado opened the pull request:
BEGINRELEASENOTES

* CLD_o3_v05 is created. This option evolves from option 2 version 05. The main characteristics with respect the mother version are:
	- ARC subdetectors included between the trackers and the ECal. 
	- Trackers are shrunk by 20 cm in radial and Z direction to accommodate the ARC. The space between layers is reduced proportionally, and the angular coverage is preserved.

ENDRELEASENOTES

Hi,

components from CLD (o2, v05) were taken to build the option 3, which contains the ARC detector, between the tracker and the ECAL. Modifications to the size of the outer tracker barrel, outer tracker endcap and inner tracker endcap were detailed in [this presentation](https://indico.cern.ch/event/1318491/contributions/5548508/attachments/2703694/4702470/ARC_IntegrationCLD_230823.pdf).

Overlap check with 10k points passed (it took 4h). 

I am doing now some efficiency plots, but I think it was a good moment to open the PR.

andresailer commented on FCCee/CLD/compact/CLD_o3_v01/InnerTracker_o2_v07.xml:
```diff
@@ -1,34 +1,34 @@
+        <constant name="InnerTracker_Barrel_half_length_2" value="692.3*mm" />
+
+        <constant name="InnerTracker_Endcap_z_0" value="524*mm" />
+        <constant name="InnerTracker_Endcap_z_1" value="773*mm" />
```
Along with the z position change we would need to adapt the inner radius accordingly to stay close to the 150 mrad coverage?

atolosadelgado replied:
thank you very much for spotting the issue!

atolosadelgado replied:
I have changed the IT endcap radius, but I am not sure if I should adapt something else, it is difficult to see by eye

andresailer commented on FCCee/CLD/compact/CLD_o3_v01/InnerTracker_o2_v07.xml:
```diff
@@ -38,34 +38,47 @@
-              <zplane z="+(Vertex_half_length + 5 * mm)"  rmin="ITEnvelopeClearanceConeHalfAngle * (Vertex_half_length + 5 * mm)" rmax="InnerTracker_outer_radius" />
-              <zplane z="+InnerTracker_half_length"       rmin="ITEnvelopeClearanceConeHalfAngle * ConeBeamPipe_zmax"             rmax="InnerTracker_outer_radius" />
-	    </shape>
+                <zplane z="-InnerTracker_half_length"
```
Is this just white-space change?

atolosadelgado replied:
it seems so, i will fix together with the change in size of the inner tracker

BrieucF commented on FCCee/CLD/compact/CLD_o3_v01/InnerTracker_o2_v07.xml:
```diff
@@ -17,13 +17,15 @@
+        <constant name="InnerTracker_Endcap_radius_3" value="192.4*mm" />
+        <constant name="InnerTracker_Endcap_radius_4" value="230.1*mm" />
+        <constant name="InnerTracker_Endcap_radius_5" value="268*mm" />
+        <constant name="InnerTracker_Endcap_radius_6" value="300.5*mm" />
```
Should'nt this change also be propagated to the sensor size?

atolosadelgado replied:
Hi, I have checked the endcaps (or disks) and it seems the change in radius makes the sensors to be placed properly, without leaving holes

andresailer replied:
One "sub-module" of a InnerTracker Module is 15.1 mm, modules are always multiples of these 15.1 mm. So reducing the radius by 30 mm leaves quite a larger gap with respect to before. Which kind of makes me afraid one would have to change the modules used. So for example instead of `InnerTrackerEndcapModule_12x8_Out` one would use a `InnerTrackerEndcapModule_12x10_Out`, which would be 30.2 mm longer. Except one would have to check that this really fits.

I am not sure if the positions before where somehow thought out that these 15.1 mm dimensions always kind of end up at the same radius, or this was just random'ish.

atolosadelgado replied:
Shall I try to replace the module size and run overlap check?

andresailer replied:
I think this needs more than just passing the overlap check, but a bit of designing.

andresailer replied:
Hi @atolosadelgado 
How is the progress with checking the numbers and gaps there are now compared to before?

atolosadelgado replied:
I just had time to run with Geantinos to assess your claim of the lack of angular coverage, but not time after Paestum to play with it. I will spend time on this as soon as I implement some new features to ARC.

atolosadelgado commented:
I just noticed that the volume names, constant names, and the readout name for the ARC do not match the style of the CLD. 

I will change it right away, I only need some minutes to do it.

atolosadelgado commented:
Changes in the naming are done, tests passed

atolosadelgado commented:
Should some release note comments be filled in the OP?

andresailer commented on FCCee/CLD/compact/CLD_o3_v05/Beampipe_o4_v05.xml:
```diff
@@ -0,0 +1,158 @@
+            <section type="Center" start="0*mm"                  end="CentralBeamPipe_zmax" rMin1="CentralBeamPipe_rmax"                  rMin2="CentralBeamPipe_rmax"                 rMax1="CentralBeamPipe_rmax+BPWWall"                rMax2="CentralBeamPipe_rmax+BPWWall"             material="AlBeMet162"    name="VertexInnerAlb" />
+            <section type="Center" start="0*mm"                  end="CentralBeamPipe_zmax" rMin1="CentralBeamPipe_rmax+BPWWall"          rMin2="CentralBeamPipe_rmax+BPWWall"         rMax1="CentralBeamPipe_rmax+BPWWall+BPWCool"        rMax2="CentralBeamPipe_rmax+BPWWall+BPWCool"     material="LiquidNDecane" name="VertexParaffin" />
+            <section type="Center" start="0*mm"                  end="CentralBeamPipe_zmax" rMin1="CentralBeamPipe_rmax+BPWWall+BPWCool"  rMin2="CentralBeamPipe_rmax+BPWWall+BPWCool" rMax1="CentralBeamPipe_rmax+2*BPWWall+BPWCool"      rMax2="CentralBeamPipe_rmax+2*BPWWall+BPWCool"   material="AlBeMet162"    name="VertexOuterAlb" />
+            <section type="Center" start="CentralBeamPipe_zmax"  end="SeparatedBeamPipe_z"  rMin1="CentralBeamPipe_rmax"                  rMin2="ConeBeamPipe_Rmax"                    rMax1="CentralBeamPipe_rmax+BeamPipeWidthFirstCone" rMax2="ConeBeamPipe_Rmax+BeamPipeWidthFirstCone" material="Beryllium"     name="AlBeMet162" />
```
```suggestion
            <section type="Center" start="CentralBeamPipe_zmax"  end="SeparatedBeamPipe_z"  rMin1="CentralBeamPipe_rmax"                  rMin2="ConeBeamPipe_Rmax"                    rMax1="CentralBeamPipe_rmax+BeamPipeWidthFirstCone" rMax2="ConeBeamPipe_Rmax+BeamPipeWidthFirstCone" material="AlBeMet162"     name="ConicalChamber" />
```

From #291

jmcarcell commented:
Can you fix the conflict @atolosadelgado? It's a very simple one line in the README

atolosadelgado commented:
I still do not understand why I get different simulation output if I use LCG, Key4hep or my local DD4hep, but since it is surely unrelated to this repository, please feel free to merge this PR at your earlier convenience :)

atolosadelgado commented:
Please, do not merge, I have just found overlap of sensor in cell 11 of the barrel... sorry for the noise

jmcarcell commented:
@atolosadelgado if you have Alma 9 the latest nightlies have Geant4 without Vecgeom. CentOS 7 will have to wait a bit and all the workflows here are CentOS 7...

atolosadelgado commented:
thank you very much! I just tested the overlaps in lxplus9 and now the test passes :):)

This is the code to reproduce the test,
```
source /cvmfs/sw-nightlies.hsf.org/key4hep/setup.sh
cd /tmp/
git clone -b CLD_with_ARC git@github.com:atolosadelgado/k4geo.git
cd k4geo/
cmake -B build -S . -D CMAKE_INSTALL_PREFIX=install
cmake --build build -j 4 -- install
export LD_LIBRARY_PATH=$PWD/install/lib:$LD_LIBRARY_PATH
cd build/
ctest -V -R t_test_ARC_o1_v01_overlap
``

At the beginning I though the problem was the unit system and I added a dummy offset in sensor position to try to fix the overlap problem. It turns out that the problem is VecGeom and that offset is not needed. I have removed that ad-hoc offset in the most recent commit. I believe this PR can be merged now.

andresailer commented:
Now there is a test failure with the overlap check

jmcarcell commented:
LCG_100 has Geant4 with Vecgeom enabled, which could explain the issues. The test with the nightlies just passed (CentOS 7 - Geant4 without Vecgeom since a while ago).

atolosadelgado commented:
Thanks Juan for the clarification. I have run the overlap test in centos 7 (LCG latest view dev4) and alma9 (LCG and key4hep) and the test passes. 
LCG 103 and before use Geant4 with VecGeom, so the overlap check will not pass.

jmcarcell commented:
Here or can we merge this PR @andresailer ? k4geo is one of the few missing tags

andresailer commented:
Yes, let's merge it and update the CI bases after
